marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.mjs

```diff
@@ -394,7 +394,23 @@ function validateAndTestMapping(mapping, uri) {
     throw Error(`Unable to validate and run mapping; could not find source document with URI '${uri}'`);
   }
 
-  const sourceDocument = cts.doc(uri);
+  let sourceDocument;
+  //modify the document instance with the properly updated one if a pre-step interceptor was used
+  let newDocumentIsLoad = false;
+  if(mapping.name){
+    const updatedDocument = invokeGetDocument(mapping.name, uri);
+    if (updatedDocument && updatedDocument.format === "JSON") {
```

Comment:
Do we need to provide alternate logic to update XML as well?

Reply:
yes, we were talking about this with Brian, and the idea is to create another ticket to handle that